ocpp/messages.py

```diff
@@ -217,9 +216,13 @@ def validate_payload(message: Union[Call, CallResult], ocpp_version: str) -> Non
             validator = get_validator(
                 message.message_type_id, message.action, ocpp_version
             )
-    except (OSError, json.JSONDecodeError):
-        raise NotImplementedError(
-            details={"cause": f"Failed to validate action: {message.action}"}
+    except OSError:
+        raise ValidationError(
+            f"JSON validation schema not found for action: {message.action}"
+        )
+    except json.JSONDecodeError:
+        raise ValidationError(
+            f"Error in decoding JSON validation schema for action: {message.action}"
```

Comment:
Only a `json.JSONDecodeError` should be mapped to a `ValidationError`. An `OSError` means that no schema with a given name was found. That means user is using a action that doesn't exists. Therefore, a `NotSupportedError` is applicable.  See my comment here: https://github.com/mobilityhouse/ocpp/pull/494#issuecomment-1786702680

Reply:
With #494 a `FileNotFoundError` should not be raised by the `OSError` exception, as before it gets to this part of the code a `NotSupportedError` will be raised by`_raise_key_error`. So an `OSError` should only be raised for one of these other reasons https://docs.python.org/3/library/exceptions.html#os-exceptions